app/cache/load.go

```diff
@@ -0,0 +1,524 @@
+package cache
+
+import (
+   "database/sql"
+   "fmt"
+   "github.com/RedHatInsights/vmaas-go/app/database"
+   "github.com/RedHatInsights/vmaas-go/app/utils"
+   "time"
+)
+
+func LoadCache() Cache {
+   c := Cache{}
+   c.Packagename2Id = loadStrIntMap("packagename", "id", "packagename", "packagename2id")
+   c.Id2Packagename = loadIntStrMap("packagename", "id", "packagename", "id2packagename")
+   c.Updates = loadKeysValsOrderedMap("updates", "name_id", "package_id",
+   "package_order", "name_id,package_order", "updates")
+   c.UpdatesIndex = loadKeysValsOrderedMap("updates_index", "name_id", "evr_id",
+   "package_order", "name_id,package_order", "updates_index")
+   c.Id2Evr, c.Evr2Id = loadEvrMaps("evr", "id", "epoch", "version", "release",
+   "id2evr, evr2id")
+   c.Id2Arch = loadIntStrMap("arch", "id", "arch", "id2arch")
+   c.Arch2Id = loadStrIntMap("arch", "id", "arch", "arch2id")
+   c.ArchCompat = loadArchCompat()
+   c.PackageDetails, c.Nevra2PkgId, c.SrcPkgId2PkgId = loadPkgDetails()
+   c.RepoDetails, c.RepoLabel2Ids, c.ProductId2RepoIds = loadRepoDetails()
+   c.PkgId2RepoIds = loadInt2Ints("pkg_repo", "pkg_id,repo_id", "pkgid2repoids") // long
+   c.ErrataDetail, c.ErrataId2Name = loadErratas("errata_detail, errataid2name")
+   c.PkgId2ErrataIds = loadInt2Ints("pkg_errata", "pkg_id,errata_id","pkgid2errataids")
+   c.ErrataId2RepoIds = loadInt2Ints("errata_repo", "errata_id,repo_id", "errataid2repoids")
+   c.CveDetail = loadCves("cve_detail")
+   c.PkgErrata2Module = loadPkgErrataModule("pkg_errata2module")
+   c.ModuleName2Ids = loadModuleName2Ids("module_name2ids")
+   c.DbChange = loadDbChanges("dbchange")
+   c.String = loadString("strings")
+   return c
+}
+
+func (c Cache) ShowInfo() {
+   fmt.Printf("%d packages\n", len(c.PackageDetails))
+}
+
+func getAllRows(tableName, cols, orderBy string) *sql.Rows {
+	rows, err := database.Db.DB().Query(fmt.Sprintf("SELECT %s FROM %s ORDER BY %s",
+       cols, tableName, orderBy))
+	if err != nil {
+		panic(err)
+	}
+	return rows
+}
+
+func loadIntArray(tableName, col, orderBy string) []int {
+   rows := getAllRows(tableName, col, orderBy)
+   defer rows.Close()
+
+   var arr []int
+   for rows.Next() {
+      var num int
+      err := rows.Scan(&num)
+      if err != nil {
+         panic(err)
+      }
+
+      arr = append(arr, num)
+   }
+   return arr
+}
+
+func loadStrArray(tableName, col, orderBy string) []string {
+   rows := getAllRows(tableName, col, orderBy)
+   defer rows.Close()
+
+   var arr []string
+   for rows.Next() {
+      var val string
+      err := rows.Scan(&val)
+      if err != nil {
+         panic(err)
+      }
+
+      arr = append(arr, val)
+   }
+   return arr
+}
+
+func loadIntStrMap(tableName, intCol, strCol, info string) map[int]string {
+   defer utils.TimeTrack(time.Now(), info)
+
+   ints := loadIntArray(tableName, intCol, intCol)
+   strs := loadStrArray(tableName, strCol, intCol)
+
```

Comment:
Hmm, this is not an optimal way to load two columns from table :).

Reply:
Yes, that's because I was looking for a way how to work with tables a little generically. Later I found out better way but this method uses this method. I am going to improve it if we will decide to use it further. Thanks for review.